Lib/fontTools/ttLib/tables/otBase.py

```diff
@@ -728,13 +728,12 @@ def compile(self, writer, font):
 					# conv.repeat is a propagated count
 					writer[conv.repeat].setValue(countValue)
 				values = value
-				for i, value in enumerate(values):
-					try:
-						conv.write(writer, font, table, value, i)
-					except Exception as e:
-						name = value.__class__.__name__ if value is not None else conv.name
-						e.args = e.args + (name+'['+str(i)+']',)
```

Comment:
hm. this is the base table's compile method, so by default we'll lose the index into the array where an exception occurs. I suppose that can sometimes be useful, e.g. which lookup or which subtable failed to be compiled for whatever reason?

Reply:
The lookup/subtable etc raise proper exception because the index (`i`) is passed down to write objects. This was a hack I introduced to get something resembling the full path down to the failure. I agree it's *sometimes* useful. I can put it back inside `writeArray`. Let me try a few things.

Then again, we were fine without this... I think I added it when I was working on 99proof.